doc/guides/naming.md

```diff
@@ -41,530 +37,352 @@ The following are **singular**:
+- `name` will still be used as the response `type`:
 
-```
-const AcmeCorp_SalesPerson = {
-  name: "SalesPerson",
-  namespace: "AcmeCorp", //👀
-  attributes: { ... }
+```js
+{
+  data: {
+    type: "SalesPerson"
+    ...
+  }
 }
 ```
 
-**Database Implications**
-
-- Creates a table sales_person in the Postgres schema acme_corp
+### namespace (Postgres only)
 
-**API Impliciations**
-
-- This will create an acme-corp/sales-persons API
-- When referencing this in the type fields, AcmeCorp_SalesPerson will be used: GET /api/acme-corp/sales-persons?fields[SalesPerson]=name
-- Data will be returned like:
-  ```
-  {
-    data: {
-      type: "AcmeCorp_SalesPerson",  // same as in "included"
-      id: "....",
-      attributes: { .... }
-    }
-  }
-  ```
+Optionally set `namespace` when using Postgres to use [Postgres Schemas](https://www.postgresql.org/docs/current/ddl-schemas.html) which are like namespaces for tables. The `namespace` must be written as singular PascalCase as follows:
 
-**Returned Models Implications**
-
-hatchifyKoa({AcmeCorp_SalesPerson}) returns models.AcmeCorp_SalesPerson
-
-### Schema.attributes.ATTRIBUTE_NAME
-
-An attribute name should be `Singular camelCase`.
-
-```js
-const SalesPerson = {
+```ts
+const AcmeCorp_SalesPerson = {
   name: "SalesPerson",
+  namespace: "AcmeCorp", //👀
   attributes: {
-    firstName: "STRING", //👀
+    firstName: string(),
   },
-}
+} satisfies PartialSchema
 ```
 
-**Database Implications**
+**Database Implications:**
 
-- Creates a column `first_name` in the `sales_person` table.
+- Creates a table `sales_person` in the Postgres schema `acme_corp`.
 
-**API Implications**
+**API Implications:**
 
-- `firstName` will be used in query parameters like
-  `GET /api/sales-persons?filter[firstName]=Mary&fields[SalesPerson]=firstName`
-- `firstName` will be used in mutation payloads and response payloads like:
+- Creates an `acme-corp/sales-persons` API.
+- `namespace_name`, will be used in the `fields` query parameter: `GET /api/acme-corp/sales-persons?fields[AcmeCorp_SalesPerson]=name`
+- `namespace_name` will be used as the response `type`:
   ```js
   {
     data: {
-      type: "SalesPerson",
-      id: "1",
-      attributes: { firstName: "Mary" } //👀
+      type: "AcmeCorp_SalesPerson", //👀
+      id: "f06f81f2-4bea-4a60-99ad-8da8ecf79473",
+      ...
     }
   }
   ```
 
-### relationships.belongsTo
+**Returned Models Implications:**
 
-A `target` option is required.
+- `hatchifyKoa({AcmeCorp_SalesPerson})` returns `models.AcmeCorp_SalesPerson`
+- `hatchifyReact({AcmeCorp_SalesPerson})` returns `[components|model|state].AcmeCorp_SalesPerson`
 
-- `target` should match a `Schema.name` and be _Singular PascalCase_.
+### id
 
-```js
-const Account = {
-  name: "Account",
-  attributes: {
-    name: "STRING",
-  },
-  belongsTo: [{ target: "SalesPerson" }], //👀
-}
-```
+JSON:API requires that the `id` attribute be named `id`, therefore this attribute cannot be renamed.
```

Comment:
not sure if this is correct @roy-coder  - i know we allow customizing the `id` via the `PartialSchema`, but can someone actually rename it? jsonapi **requires** for id to be present in responses, are we enforcing this on hatchify as well?

Reply:
I will allow @roy-coder to confirm, but AFAIK yes we require `id` to be present in all returned records.